.claude/scripts/egress-third-party-qualifier-contract.test.sh

```
@@ -0,0 +1,129 @@
+esac
+
+# 3. PRESERVATION — both gates must sit inside ONE CONTIGUOUS affirmative clause.
+has 'only once both its gates are cleared** — the professional-work boundary and the explicit per-artifact approval' "${egress}" || \
+  fail "the third-party gate's two requirements are no longer bound inside one contiguous affirmative clause"
+ok
+
+# 4. The exemption must follow the devantler-tech OWNER, never the word "upstream". *Definition
+#    routing* calls a synced skill's repository an upstream too, and those are frequently third party
+#    (`find-skills` is owned by `vercel-labs/skills`) — so an exemption phrased as "the skills
+#    repositories" would exempt a third-party owner from the gate this very entry imposes.
+has 'follows the `devantler-tech` owner' "${egress}" || \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)